ssl/statem/extensions_clnt.c

```diff
@@ -1452,14 +1452,8 @@ int tls_parse_stoc_status_request(SSL *s, PACKET *pkt, unsigned int context,
     }
 
     if (SSL_IS_TLS13(s)) {
```

Comment:
There should probably be a run-time option to ignore anything beyond the leaf status.

Reply:
The client reads all OCSP responses sent by the server. The verifying of the response was handled by the flags "crl_check" and "crl_check_all". 
In the last commit, I added two flags "ocsp_check" and "ocsp_check_all" to verify the OCSP response only of the leaf certificate or of all certificates in the chain. I also added these flags in the man page.

Reply:
I'm actually not sure yet of what we want. I think it makes sense to verify all the OCSP responses that were received. And the difference is that we only require one for the leave, or also for the intermediates. But then I'm not sure that "require" is the correct word, I think we only request one, and if we don't get it, it's up to the application to deal with it?

Reply:
I guess the existing callback can just be used to see if OCSP was stapled for that certificate or not. If the X509_V_ERR_OCSP_NO_RESPONSE error was returned, it wasn't present. I don't think there is a need the ocsp_check_all, the application can just deal with that in the call back. The only difference between the ocsp_check and ocsp_check_all would be that the intermediates get that error if it's not received.

But I'm not sure I find that a good API or not.